cupy/cuda/compiler.py

```diff
@@ -253,7 +258,15 @@ def _compile(
 
         prog = _NVRTCProgram(source, cu_path, headers, include_names,
                              name_expressions=name_expressions)
+
+        options += (_get_arch_for_options(arch=arch, jitify=jitify),)
+
         try:
+            if _cuda_hip_version >= 11010 and jitify:
+                # Convert the virtual arch to a real arch
```

Comment:
Isn't this already done in `_get_arch_for_options`?

Reply:
this is some kind of tricky case when jitify is used.

Reply:
maybe @leofang can explain it better :)

Reply:
Now, by tracking the method, we can also make it completely `jitify` independent :), so we can get rid of these hacks too.